### PR TITLE
Add new Azure resource pack backend dependencies

### PR DESCRIPTION
**Pull request by omerdemirok**
Signed-off-by: Omer Demirok <odemirok@chef.io>

## Description
The new Azure resource pack [backend](https://github.com/inspec/inspec-azure/tree/azure_generic_resources) requires following additional dependencies to be added:
- [Dotenv](https://github.com/bkeepers/dotenv)
- [ipaddr](https://github.com/ruby/ipaddr)
- [facets](https://github.com/rubyworks/facets)
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by omerdemirok** [comment]
CC: @rmoles, @skpaterson

File: train.gemspec [hunk 1]
```diff
@@ -41,4 +41,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "docker-api", "~> 1.26"
   spec.add_dependency "google-api-client", ">= 0.23.9", "< 0.35.0"
   spec.add_dependency "googleauth", ">= 0.6.6", "< 0.11.0"
+  spec.add_dependency "dotenv", "~> 2.7.6"
+  spec.add_dependency "facets", "~> 3.1.0"
+  spec.add_dependency "ipaddr", "~> 1.2"
```
Comment (clintoncwolfe) [inline train.gemspec:46]: A small ask: maybe a comment noting that these are required for azure, or group them into their own section using a comment and some whitespace?

File: train.gemspec [hunk 2]
```diff
@@ -41,4 +41,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "docker-api", "~> 1.26"
   spec.add_dependency "google-api-client", ">= 0.23.9", "< 0.35.0"
   spec.add_dependency "googleauth", ">= 0.6.6", "< 0.11.0"
+  spec.add_dependency "dotenv", "~> 2.7.6"
+  spec.add_dependency "facets", "~> 3.1.0"
```
Comment (tas50) [inline train.gemspec:45]: Is this something we really need or are we just pulling these in so we can use some helpers? Train gets installed in a lot of places and this doesn't seem like a must have for implemented azure support.

Comment (omerdemirok) [reply train.gemspec:45]: Working on removing `facets` and `dotenv`. However, `ipaddr` is essential to make security rules comparison work properly. It is provided in the AWS resource pack.

**Comment by omerdemirok** [comment]
Found out that [`ipaddr`](https://github.com/ruby/ipaddr#installation) is shipped in Ruby distros. 
There is no need to add a new dependency at this point.
Thanks, @clintoncwolfe, @tas50, and @Schwad.

**Comment by Schwad** [comment]
Thanks for your work in this area @omerdemirok ! Much appreciated 🙏

**Comment by tas50** [comment]
I'm glad it turned out we didn't need this one. Keeping this dep list as small as possible is an important goal.

**Pull request closed without merging**